src/split-props.js

```diff
@@ -11,21 +11,22 @@ export default function splitProps(
     // component ever
     ...rest
   },
-  {propsAreCssOverrides, rootEl, forwardProps},
+  {propsAreCssOverrides, rootEl, filterProps, forwardProps},
 ) {
   const returnValue = {toForward: {}, cssProp, cssOverrides: {}}
   if (!propsAreCssOverrides) {
-    if (typeof rootEl !== 'string') {
-      // if it's not a string, then we can forward everything
-      // (because it's a component)
+    if (typeof rootEl !== 'string' && filterProps.length === 0) {
+      // if it's not a string and filterProps is empty,
+      // then we can forward everything (because it's a component)
       returnValue.toForward = rest
       return returnValue
     }
   }
   return Object.keys(rest).reduce((split, propName) => {
     if (
-      forwardProps.indexOf(propName) !== -1 ||
-      shouldForwardProperty(rootEl, propName)
+      filterProps.indexOf(propName) === -1 &&
+      (forwardProps.indexOf(propName) !== -1 ||
+        shouldForwardProperty(rootEl, propName))
     ) {
       split.toForward[propName] = rest[propName]
     } else if (propsAreCssOverrides) {
```

Comment:
Could you add a test that takes `propsAreCssOverrides` into account? I have a feeling that this logic should change to: `filterProps.indexOf(propName) === -1 && propsAreCssOverrides`. With that in mind, we should probably make a variable: `const shouldFilter = filterProps.indexOf(propName) !== -1` and use that.

Reply:
To be sure you mean
```javascript
} else if (filterProps.indexOf(propName) === -1 && propsAreCssOverrides) {
```
instead of

```javascript
} else if (propsAreCssOverrides) {
```
?

I am currently not sure if
```javascript
glamorous('div', {
  filterProps: ['margin'],
  propsAreCssOverrides: true,
})({
  margin: 1,
  padding: 1,
})
```
makes sense at all.

Reply:
What about:

```javascript
const Styled = glamorous(MyComponent, {
  rootEl: 'div',
  filterProps: ['faded'],
  propsAreCssOverrides: true,
})(
  ({faded}) => ({/* styles */}) : null
)

const ui = <Styled faded>Hi</Styled>
```

Reply:
Yes indeed. That's a complete new use case for filterProps :smile: Currently faded would land in the CSS.